reactioncommerce/reaction
# Fix upload image error for merchant store owner

Resolves #2888 

Change Introduced:
* Ensure that userId is passed into `getShopId()`. UserId is passed to getShopId to ensure that it returns the correct shop based on the User Preference. If not passed, getShopId can default to primaryShop (hence why it has been working for the primary shop owner). 

What else might be affected by this change:
* The above change was made for the Security's defined method for `ifHasRoleForActiveShop` and `ifFileBelongsToShop` (see diff). This is place on a few other collections (part from Media), I tested around the app to confirm no side effects. 
And based on the way [getShopId is setup](https://github.com/reactioncommerce/reaction/blob/marketplace/server/api/core/core.js#L329-L338), it first tries to use it's internal activeUserId first before falling to the passed in one. So if the userId changed here isn't available, it's previous setup should still be intact.
Please raise concerns if any.



**How To Test:**
* Create a merchant store (either by logging in as guest and using "Become A Seller button" or via Dashboard > Shop > Marketplace in primary shop).
* While logged into the merchant store as it's owner, create product and add an image to that product.
* Confirm the upload works and image is uploaded successfully.
